### DISCUSSION TITLE
Grafana image render plugin

### DISCUSSION THREAD

**Discussion by njohnson-q** [Security-Onion-Solutions/securityonion#6413] [Unsupported Versions] [answered]
Tested in version 2.3.70 and 2.3.90
On prem deployment
Existing install (2.3.70) and fresh install (2.3.90)
1 manager node and 1 search node
so-status shows all green
No issues with salt
SOC Grid show no errors

We are trying to setup some reporting functions with a 3rd party script that connects to grafana and creates a pdf report of the dashboard. One dependency is using the grafana image render plugin. I can install this easily but any time I try to render an image (tested with and without third party script) it errors with the log below. I already installed the dependency it asks and it still errors.

I believe that it may be because the docker container does not have chrome installed. Is there a way that I could add this into the container?

Has anyone else tried to install this plugin?

t=2021-12-01T15:21:48+0000 lvl=eror msg="Failed to get browser version" logger=plugins.backend pluginId=grafana-image-renderer err=map[]
t=2021-12-01T15:22:06+0000 lvl=eror msg="Render request failed" logger=plugins.backend pluginId=grafana-image-renderer url="http://localhost:3000/d-solo/so_overview/security-onion-grid-overview?orgId=1&refresh=5m&from=1638361324859&to=1638372124859&panelId=69016&width=1000&height=500&tz=America%2FNew_York&render=1" error="Error: Failed to launch the browser process!\n/var/lib/grafana/plugins/grafana-image-renderer/chrome-linux/chrome: error while loading shared libraries: libgobject-2.0.so.0: cannot open shared object file: No such file or directory\n\n\nTROUBLESHOOTING: https://github.com/puppeteer/puppeteer/blob/main/docs/troubleshooting.md\n"
t=2021-12-01T15:22:06+0000 lvl=eror msg="Rendering failed." logger=context userId=1 orgId=1 uname=admin error="rendering failed: Error: Failed to launch the browser process!\n/var/lib/grafana/plugins/grafana-image-renderer/chrome-linux/chrome: error while loading shared libraries: libgobject-2.0.so.0: cannot open shared object file: No such file or directory\n\n\nTROUBLESHOOTING: https://github.com/puppeteer/puppeteer/blob/main/docs/troubleshooting.md\n"
t=2021-12-01T15:22:06+0000 lvl=eror msg="Request Completed" logger=context userId=1 orgId=1 uname=admin method=GET path=/render/d-solo/so_overview/security-onion-grid-overview status=500 remote_addr=IPADDRESS time_ms=98 size=1352 referer=

**Comment by dougburks** [comment, marked as answer]
The logs mention puppeteer. Perhaps you could accomplish your goal using puppeteer on a host outside of Security Onion, so that you wouldn't need to modify Security Onion or its containers.

**Reply by njohnson-q** [reply]
What we ended up doing is adding the /usr/lib64 to the init.sls file and set it to /usr/lib for the grafana container. I know this will be removed as grafana updates but this did work for our setup and can work with that.
